prog/hints_test.go

```diff
@@ -55,10 +55,92 @@ func TestHintsCheckConstArg(t *testing.T) {
 			[]uint64{0x2},
 		},
 	}
-	for _, test := range tests {
+	runHintsConstArgTests(tests, t, []uint64{4, 2, 2})
+}
+
+// Checks if new value fits into field during hints mutation. If the replacer
+// has a higher value, it will be truncated.
+func TestHintsCheckSizeConstArg(t *testing.T) {
+	t.Parallel()
+	var tests = []ConstArgTest{
+		{
+			"Int8 invalid value (positive)",
+			0x1234,
+			CompMap{
+				// void test8(i8 el) {
+				//		i16 w = (i16) el
+				//		if (w == 0x88) {...}
+				//		i16 other = 0xfffe
+				// 		if (w == other)
+				//  }; test8(i8(0x1234));
+				0x34: uint64Set{0x88: true, 0x1122: true, 0xfffffffffffffffe: true},
+				// This key should be iggnored.
+				0x1234: uint64Set{0xa1: true},
+			},
+			[]uint64{0x88, 0xfe},
+		},
+		{
+			"Int8 invalid value (negative)",
+			0x12ab,
+			CompMap{
+				0xab:               uint64Set{0xab: true, 0xac: true, 0xabcd: true},
+				0xffffffffffffffab: uint64Set{0x11: true, 0x22: true, 0xffffffffffffff34: true},
+			},
+			[]uint64{0x11, 0x22, 0x34, 0xac},
+		},
+		{
+			"Int16 valid value (bitsize=11)",
+			0x3ab,
+			CompMap{
+				0x3ab:              uint64Set{0x11: true, 0x1234: true, 0xfffffffffffffffe: true},
+				0xffffffffffffffab: uint64Set{0xfffffffffffffff1: true},
+			},
+			[]uint64{0x11, 0x3f1, 0x7fe},
+		},
+		{
+			"Int16 invalid value (bitsize=11)",
+			0x71ab,
+			CompMap{
+				0x1ab: uint64Set{0x11: true, 0x1234: true, 0xfffffffffffffffe: true},
+			},
+			[]uint64{0x11, 0x7fe},
+		},
+		{
+			"Int16 valid value (negative)",
+			0xffffffffffffffcd,
+			CompMap{0xffffffffffffffcd: uint64Set{0xab: true, 0xabcd: true, 0xcafebabe: true, 0xffffffffffffcafe: true}},
+			[]uint64{0xab, 0xabcd, 0xcafe, 0xffab},
+		},
+		{
+			"Int32 invalid value",
+			0xaabaddcafe,
+			CompMap{0xbaddcafe: uint64Set{0xab: true, 0xabcd: true, 0xbaddcafe: true,
+				0xdeadbeef: true, 0xaabbccddeeff1122: true}},
+			[]uint64{0xab, 0xabcd, 0xdeadbeef},
+		},
+		{
+			"Int64 valid value",
+			0xdeadc0debaddcafe,
+			CompMap{0xdeadc0debaddcafe: uint64Set{0xab: true, 0xabcd: true, 0xdeadbeef: true, 0xdeadbeefdeadbeef: true}},
+			[]uint64{0xab, 0xabcd, 0xdeadbeef, 0xdeadbeefdeadbeef},
+		},
+	}
+	runHintsConstArgTests(tests, t, []uint64{1, 1, 2, 2, 2, 4, 8}, []uint64{0, 0, 11, 11, 0, 0, 0})
```

Comment:
These arrays absolutely need to be members of ConstArgTest.
Say, I am looking at 11-th test case. Now to understand what it does, I need to count 11-th integer in an array somewhere screens below, then count another 11-th element in another parallel array, then assuming I don't mess counting, I need to go up screens and find the test again, at this point I may have forgotten what test I was looking at, and then when I remember it I forget the number in the first array, so I need to repeat this...
This data totally needs to be local with the test case. ConstArgTest specifically encapsulates all inputs data for a test, so that it's local.
Now that we have 2 optional arguments and we need to refactor it anyway, maybe it's time to add names to struct fields and then we can omit the optional ones.

Also you don't need to factor out common code into runHintsConstArgTests and create another test. That's the advantage of table-driven tests, you can just keep adding more tests to the existing table. Especially since we have a name for each test, so it's easy to map test failure to the test case.